pandas/core/indexes/base.py

```diff
@@ -3270,7 +3270,9 @@ def _can_reindex(self, indexer):
         """
         # trying to reindex on an axis with duplicates
         if not self.is_unique and len(indexer):
-            raise ValueError("cannot reindex from a duplicate axis")
+            raise ValueError("""cannot reindex from a duplicate axis
+        (this usually occurs when trying to join / assign
+        to a column when the index contains duplicate values).""")
```

Comment:
Can you assign this to something called msg then reference within the ValueError? This shows the preferred pattern for this kind of thing:

https://github.com/pandas-dev/pandas/blob/9820c426a035495ca14dfb0fefe873e96f7483e0/pandas/core/reshape/merge.py#L1696

Also please run black before as well to make sure the changes pass linting.

Reply:
I'm not familiar with this term - does `run black` mean to pass all checks?

Reply:
black is the tool pandas uses for code formatting, this link should help but let me know if you have any other questions:

https://pandas.pydata.org/docs/dev/development/contributing.html#python-pep8-black